unit-tests/src/test/scala/java/nio/file/FilesSuite.scala

```diff
@@ -891,6 +894,64 @@ object FilesSuite extends tests.Suite {
     }
   }
 
+  // This test was inspired by Issue #1354
+  // "NIO File Walker fails on broken links."
+  test("Files.walkFileTree respects FileVisitOption.FOLLOW_LINKS") {
+    withTemporaryDirectory { dirFile =>
+      val dir = dirFile.toPath()
+      val f0  = dir.resolve("f0")
+      val f1  = dir.resolve("f1")
+
+      val brokenLink    = dir.resolve("brokenlink")
+      val missingTarget = dir.resolve("missingtarget")
+
+      Files.createFile(f0)
+      Files.createFile(f1)
+
+      assert(Files.exists(f0) && Files.isRegularFile(f0))
+      assert(Files.exists(f1) && Files.isRegularFile(f1))
+
+      Files.createFile(missingTarget)
+
+      assert(
+        Files.exists(missingTarget) &&
+          Files.isRegularFile(missingTarget))
+
+      // Create valid symbolic link from brokenLink to missingTarget,
+      // then remove missingTarget to break link.
+
+      Files.createSymbolicLink(brokenLink, missingTarget)
```

Comment:
Is it necessary to create the missingTarget to create the symbolicLink? I was under the impression that the api allows one to create broken symbolic links.

Reply:
This is probably superstition and/or lack of knowledge on my part. I was trying to 
take the same steps as would probably be taken in Real Life and as reported
in the original issue.

I can change it if you like or think it is misdirection that would mislead others or
myself six months down the line.

Please advise.

Reply:
I don't really care.

Reply:
I left it the way it was.  Superstition on my part. File systems have a way of merging
operations or converting them to no-ops underneath.  
Thank you.